TidepoolServiceKitUI/SettingsView.swift

```diff
@@ -0,0 +1,218 @@
+                            VStack {
+                                Text(LocalizedString("Environment", comment: "Label title for displaying selected Tidepool server environment."))
+                                    .bold()
+                                Text(selectedEnvironment.description)
+
+                                if isLoggedIn {
+                                    Button(LocalizedString("Revoke token", comment: "Button title to revoke oauth tokens"), action: {
+                                        Task {
+                                            do {
+                                                try await service.tapi.revokeTokens()
+                                            } catch {
+                                                self.error = error
+                                            }
+                                        }
+                                    })
+                                }
+                            }
+                        }
+                        if let username = service.session?.username {
+                            VStack {
+                                Text(LocalizedString("Logged in as", comment: "LoginViewModel description text when logged in"))
+                                    .bold()
+                                Text(username)
+                            }
+                        } else {
+                            Text(LocalizedString("You are not logged in.", comment: "LoginViewModel description text when not logged in"))
+                                .padding()
+                        }
+
+                        if let error {
+                            VStack(alignment: .leading) {
+                                Text(error.localizedDescription)
+                                    .font(.callout)
+                                    .foregroundColor(.red)
+                            }
+                            .padding()
+                        }
+                        Spacer()
+                        if isLoggedIn {
+                            deleteServiceButton
+                        } else {
+                            loginButton
+                        }
+                    }
+                    .padding()
+                    .frame(minHeight: geometry.size.height)
+                }
+            }
+        }
+        .alert(LocalizedString("Are you sure you want to delete this service?", comment: "Confirmation message for deleting a service"), isPresented: $showingDeletionConfirmation)
+        {
+            Button(LocalizedString("Delete Service", comment: "Button title to delete a service"), role: .destructive) {
+                service.deleteService()
+                dismiss?()
+            }
+        }
+        .task {
+            do {
+                environments = try await TEnvironment.fetchEnvironments()
+            } catch {
+
```

Comment:
Do we need to write to the error property here?
```suggestion
self.error = error
```

Reply:
No, the environments is a "secret" list, that shouldn't expose errors to the user on failure. But good question.